libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CloudAdapter.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// An adapter that implements the Bot Framework Protocol and can be hosted in different cloud environmens both public and private.
+    /// </summary>
+    public class CloudAdapter : CloudAdapterBase, IBotFrameworkHttpAdapter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudAdapter"/> class. (Public cloud. No auth. For testing.)
+        /// </summary>
+        public CloudAdapter()
+            : base(BotFrameworkAuthenticationFactory.Create(null, false, null, null, null, null, null, null, null, new PasswordServiceClientCredentialFactory(), new AuthenticationConfiguration(), null, null))
```

Comment:
Try calling the base from only one place so it is easier to track the code path, the suggestion below should work:

```suggestion
            : this(BotFrameworkAuthenticationFactory.Create(null, false, null, null, null, null, null, null, null, new PasswordServiceClientCredentialFactory(), new AuthenticationConfiguration(), null, null))
```